### PR TITLE
ML-4799: Simplify and optimize emit source offset handling for performance

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: storey/sources.py [hunk 1]
```diff
@@ -249,6 +250,7 @@ class SyncEmitSource(Flow):
     :param buffer_size: size of the incoming event buffer. Defaults to 8.
     :param key_field: Field to extract and use as the key. Optional.
     :param max_events_before_commit: Maximum number of events to be processed before committing offsets.
+      Defaults to 10,000.
```
Comment (assaf758) [inline storey/sources.py]: 20000
also please add description of max_time_before_commit

File: storey/sources.py [hunk 2]
```diff
@@ -294,36 +297,41 @@ async def _run_loop(self):
...
-                    can_block = await _commit_handled_events(self._outstanding_offsets, committer)
-            if not event:
+            if committer:
+                if (
+                    events_handled_since_commit >= self._max_events_before_commit
+                    or num_offsets_not_handled > 1
+                    and time.monotonic() >= last_commit_time + self._max_time_before_commit
+                ):
+                    num_offsets_not_handled = await _commit_handled_events(self._outstanding_offsets, committer)
+                    events_handled_since_commit = 0
```
Comment (assaf758) [inline storey/sources.py:314]:
```suggestion
                    events_handled_since_last_commit = 0
```

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: storey/sources.py [hunk 3]
```diff
@@ -580,37 +590,40 @@ def _init(self):
...
+                    last_commit_time = time.monotonic()
             if not event:
                 event = await self._q.get()
             if committer and hasattr(event, "path") and hasattr(event, "shard_id") and hasattr(event, "offset"):
                 qualified_shard = (event.path, event.shard_id)
                 offsets = self._outstanding_offsets[qualified_shard]
                 offsets.append(_EventOffset(event))
-                num_events_handled_without_commit += 1
+                num_offsets_not_handled += 1
```
Comment (gtopper) [inline storey/sources.py:627]:
```suggestion
                num_offsets_not_committed += 1
```

File: storey/sources.py (hunk 1 above)
Comment (assaf758) [reply storey/sources.py]: you missed adding description of max_time_before_commit

**Review by assaf758 — APPROVED** [review]

**Pull request merged**